app/lib/features/chat/utils.dart

```diff
@@ -244,3 +246,32 @@ String prepareMsg(MsgContent? content) {
     (match) => '<a href="${match.group(0)}">${match.group(0)}</a>',
   );
 }
+
+String parseEditMsg(WidgetRef ref) {
+  // shouldn't ever happen to be null if state is edit or reply
+  final message = ref.read(chatInputProvider).selectedMessage!;
```

Comment:
may I suggest a change of API here...

```suggestion
String parseEditMsg(Message message) {
```

and also, as this is then nicely easily input-output-based ... can we have unit tests for this confirming it does what we want? Should be rather straight forward then.